### PR TITLE
update model workspace CTA to point to pre-existing dashboard(s) instead of "create new dashboard"

### LINKED ISSUE #1622: model flow: if there's an existing dashboard, change CTA to select that instead of offering to create a new one

### PR DESCRIPTION
**Pull request by hamilton**
closes #1622

This PR aims to do the minimum amount necessary to update the CTA on the modeling page. It's part of a larger set of changes involved for #1620. This PR does the following:
- Implements a tweak to the main `Button.svelte` component, needed for additional changes in #1620
- if the model already has a dashboard associated with it, replace the `Create Dashboard` button with `Go to Dashboard`
- if the model has a few dashboards, the CTA will actually drop down a menu.
- adds a new `Forward.svelte` icon from our figma icon set

### CODE REVIEW

File: web-common/src/features/models/workspace/CreateDashboardButton.svelte [hunk 1]
```diff
@@ -40,6 +41,7 @@
     modelName
   );
   $: model = $getModel.data?.entry?.model;
+  $: timestampColumns = selectTimestampColumnFromSchema(model?.schema);
```
Comment (djbarnwal) [inline web-common/src/features/models/workspace/CreateDashboardButton.svelte:44]: We don't need this variable (see below)

File: web-common/src/features/models/workspace/CreateDashboardButton.svelte [hunk 2]
```diff
@@ -101,11 +103,17 @@
...
+<Tooltip alignment="right" distance={8} location="bottom">
+  <Button
+    disabled={!timestampColumns?.length}
+    on:click={handleCreateDashboard}
+    type="primary"
   >
+    <IconSpaceFixer pullLeft pullRight={collapse}>
+      <Add />
+    </IconSpaceFixer>
+    <ResponsiveButtonText {collapse}>Create Dashboard</ResponsiveButtonText>
```
Comment (djbarnwal) [inline web-common/src/features/models/workspace/CreateDashboardButton.svelte:115]: There is no tooltip content for the Create Dashboard button. 
This is also present in main but probably best to handle in this PR.

image

File: web-common/src/features/models/workspace/ModelWorkspaceHeader.svelte [hunk 3]
```diff
@@ -45,12 +41,19 @@
...
+    {
+      query: {
+        select(data) {
+          return data?.entries?.filter(
+            (entry) => entry?.metricsView?.model === modelName
+          );
+        },
+      },
+    }
+  );
```
Comment (djbarnwal) [inline web-common/src/features/models/workspace/ModelWorkspaceHeader.svelte:56]: We can move this to `svelte-query/dashbords.ts` (which will be later moved to `features/dashboards/selectors.ts`)

File: web-common/src/features/sources/workspace/SourceWorkspaceHeader.svelte [hunk 4]
```diff
@@ -288,10 +296,13 @@
               on:click={() => handleCreateDashboardFromSource(sourceName)}
               type="primary"
             >
+              <IconSpaceFixer pullLeft pullRight={width < 800}
```
Comment (djbarnwal) [inline web-common/src/features/sources/workspace/SourceWorkspaceHeader.svelte:299]: Nit: we can move these multiple `width < 800` inside the script tag.

**Review by djbarnwal — CHANGES_REQUESTED** [review]
I am unable to create a dashboard out of a model with no time dimensions. Apart from that looks good, have added a few nit comments.

**Review by djbarnwal — APPROVED** [review]

**Comment by ericpgreen2** [comment]
Code looks good.

Experience-wise, I think it'd be good to retain a CTA to create a new dashboard, alongside the CTA to go to an existing one. Scott says it's a common pattern for our customers to have multiple dashboards for one model (where often fwiw each dashboard is a different subset of measures/dimensions). I know it's _technically_ possible to still create two dashboards, but it's not obvious.

**Comment by hamilton** [comment]
It's a good point re: multiple dashboards on one model, but I think this will be the way we meet the underlying need going forward. There are many other ways to solve this issue that don't look like child dashboards, including RBAC and other affordances that aren't in scope for Rill Developer. Given that this is not a one-way door I would prefer we merge as-is and we can readdress later. Thanks for the review @ericpgreen2!

**Pull request merged**